### PR TITLE
Raidboss: Add common replacement text for directions and roles

### LINKED ISSUE #1821: Raidboss: Directions in common replacement?
## Summary

There are many timeline using directions like `(NW)` `(SE)` `(NE/SE)`, to specify where to handle the mechanic.

But then we should write a bunch of replacement text to localize it.

Just like:
```javascript
'\\(NE\\)': '(北東)',
'\\(NW\\)': '(北西)',
'\\(SE\\)': '(南東)',
'\\(SW\\)': '(南西)',
```

So, can we just put them into `common_replacement.js`?

Also, maybe some role-specific text like `(T)` `(DPS)`, `(T/H)`, can be put into that file, too?

### PR DESCRIPTION
**Pull request by MaikoTan**
Mentioned in #1821 

- Initial commit: add directions matching regex

- Closes #1821

### CODE REVIEW

File: ui/raidboss/common_replacement.js [hunk 1]
```diff
@@ -9,6 +9,24 @@ const syncKeys = {
...
+  W: '(?<= \\(|\\/)W(?=\\)|\\/)',
+  NE: '(?<= \\(|\\/)NE(?=\\)|\\/)',
+  NW: '(?<= \\(|\\/)NW(?=\\)|\\/)',
+  SE: '(?<= \\(|\\/)SE(?=\\)|\\/)',
+  SW: '(?<= \\(|\\/)SW(?=\\)|\\/)',
+  // Match Roles in replaceText
+  // eg: `(Tank)`, `(Healer)`, `(DPS)`, etc
+  Tank: '(?<= \\(|\\/)Tank(?=\\)|\\/)',
+  Healer: '(?<= \\(|\\/)Healer(?=\\)|\\/)',
+  DPS: '(?<= \\(|\\/)DPS(?=\\)|\\/)',
```
Comment (MaikoTan) [inline ui/raidboss/common_replacement.js:27]: Maybe it is not very useful to translate something like `(T)` and `(T/H)`?
So I have only added full names of roles.

File: ui/raidboss/common_replacement.js [hunk 2]
```diff
@@ -204,6 +222,83 @@ const commonReplacement = {
...
+    [textKeys.Tank]: {
+      de: 'Tank',
+      fr: 'Tank',
+      ja: 'タンク',
+      cn: '坦克',
+      ko: '방어',
+    },
+    [textKeys.Healer]: {
+      de: 'Heiler',
+      fr: 'Soigneurs',
```
Comment (VanSlanzar) [inline ui/raidboss/common_replacement.js:290]: Hi, no translate plz ^^. Just `Healer`.

Comment (MaikoTan) [reply ui/raidboss/common_replacement.js]: OK, I have changed it back

**Comment by Jaehyuk-Lee** [comment]
Korean east, south, ... it'll be okay to remove `쪽`, it means `-ward`.
ex) `동쪽` is `eastward`. `동` is `east` but it's very unnatural in sentence, this is why translator or many example sentences don't use it alone. People understand its meaning. for reference, `東` is `동`. Same as `東` is `ひがし`. (Korean always write in its pronounce.)

File: ui/raidboss/common_replacement.js [hunk 3]
```diff
@@ -204,6 +222,83 @@ const commonReplacement = {
...
+      ja: '南西',
+      cn: '西南',
+      ko: '남서',
+    },
+    [textKeys.Tank]: {
+      de: 'Tank',
+      fr: 'Tank',
+      ja: 'タンク',
+      cn: '坦克',
+      ko: '방어',
```
Comment (Jaehyuk-Lee) [inline ui/raidboss/common_replacement.js:286]: ko: '탱커',

File: ui/raidboss/common_replacement.js [hunk 4]
```diff
@@ -204,6 +222,83 @@ const commonReplacement = {
...
+      ja: 'ヒーラー',
+      cn: '治疗',
+      ko: '힐러',
+    },
+    [textKeys.DPS]: {
+      de: 'DPS',
+      fr: 'DPS',
+      ja: 'DPS',
+      cn: 'DPS',
+      ko: 'DPS',
```
Comment (Jaehyuk-Lee) [inline ui/raidboss/common_replacement.js:300]: ko: '딜러',

File: ui/raidboss/common_replacement.js [hunk 5]
```diff
@@ -204,6 +222,83 @@ const commonReplacement = {
       cn: '--无法选中--',
       ko: '--타겟 불가능--',
     },
+    [textKeys.E]: {
+      de: 'O',
+      fr: 'E',
+      ja: '東',
+      cn: '东',
+      ko: '동쪽', // FIXME: looks like Korean has no abbrivate of "east"?
```
Comment (Jaehyuk-Lee) [inline ui/raidboss/common_replacement.js:230]: ko: '동',

File: ui/raidboss/common_replacement.js [hunk 6]
```diff
@@ -204,6 +222,83 @@ const commonReplacement = {
...
+    },
+    [textKeys.N]: {
+      de: 'N',
+      fr: 'N',
+      ja: '北',
+      cn: '北',
+      ko: '북쪽', // FIXME
```
Comment (Jaehyuk-Lee) [inline ui/raidboss/common_replacement.js:237]: ko: '북',

File: ui/raidboss/common_replacement.js [hunk 7]
```diff
@@ -204,6 +222,83 @@ const commonReplacement = {
...
+    },
+    [textKeys.S]: {
+      de: 'S',
+      fr: 'S',
+      ja: '南',
+      cn: '南面',
+      ko: '남쪽', // FIXME
```
Comment (Jaehyuk-Lee) [inline ui/raidboss/common_replacement.js:244]: ko: '남',

File: ui/raidboss/common_replacement.js [hunk 8]
```diff
@@ -204,6 +222,83 @@ const commonReplacement = {
...
+    },
+    [textKeys.W]: {
+      de: 'W',
+      fr: 'O',
+      ja: '西',
+      cn: '西',
+      ko: '서쪽', // FIXME
```
Comment (Jaehyuk-Lee) [inline ui/raidboss/common_replacement.js:251]: ko: '서',

**Comment by MaikoTan** [comment]
Thanks for explaining. I have fixed those words.

**Comment by MaikoTan** [comment]
I have changed files with directions that I have added, but I don't know how to deal with the following error.
```
/home/***/workspace/cactbot/ui/raidboss/data/02-arr/raid/t4.js:locale de: pre-translation collision on replaceText '2x Rook (E/W center)' for '\(E/W center\)' and '(?<= \(|\/)E(?=\)|\/)'
/home/***/workspace/cactbot/ui/raidboss/data/02-arr/raid/t4.js:locale de: pre-translation collision on replaceText '2x Rook (E/W center)' for '(?<= \(|\/)E(?=\)|\/)' and '\(E/W center\)'
/home/***/workspace/cactbot/ui/raidboss/data/02-arr/raid/t4.js:locale ja: pre-translation collision on replaceText '2x Rook (E/W center)' for '\(E/W center\)' and '(?<= \(|\/)E(?=\)|\/)'
/home/***/workspace/cactbot/ui/raidboss/data/02-arr/raid/t4.js:locale ja: pre-translation collision on replaceText '2x Rook (E/W center)' for '(?<= \(|\/)E(?=\)|\/)' and '\(E/W center\)'
/home/***/workspace/cactbot/ui/raidboss/data/02-arr/raid/t4.js:locale cn: pre-translation collision on replaceText '2x Rook (E/W center)' for '\(E/W center\)' and '(?<= \(|\/)E(?=\)|\/)'
/home/***/workspace/cactbot/ui/raidboss/data/02-arr/raid/t4.js:locale cn: pre-translation collision on replaceText '2x Rook (E/W center)' for '(?<= \(|\/)E(?=\)|\/)' and '\(E/W center\)'
The terminal process "/bin/bash '-c', 'python test/run_tests.py'" terminated with exit code: 1.
```

The related files are: https://github.com/quisquous/cactbot/blob/d36dd2b545f3e4e9ea87d5b4a001a0f8c0a420e2/ui/raidboss/data/02-arr/raid/t4.txt#L12

https://github.com/quisquous/cactbot/blob/d36dd2b545f3e4e9ea87d5b4a001a0f8c0a420e2/ui/raidboss/data/02-arr/raid/t4.js#L63-L172

**Comment by Jaehyuk-Lee** [comment]
The problem is that `(?<= \(|\/)E(?=\)|\/)` catches `E` in `(E/W center)`.
but there's `'\\(E/W center\\)': '(O/W mitte)',` replacement. So, two collides.

You should change
`'\\(E/W center\\)': '(O/W mitte)',` to `'center': 'mitte',`
but there's already replacement for `'\\(center\\)'`, so just remove every `'\\(E/W center\\)'` replacements, then change `'\\(center\\)'` to `center`.

### example
```javascript
      'replaceText': {
        '\\(E/W center\\)': '(O/W mitte)',
        '\\(SE/NW\\)': '(SO/NW)',
        '\\(center\\)': '(mitte)',
        '\\(outside\\)': '(draußen)',
        'Bug': 'Wanze',
        'Dreadnaught': 'Brummonaut',
        'Emergency Override': 'Not-Übersteuerung',
        'Knight': 'Ritter',
        'Rook': 'Drehturm',
        'Soldier': 'Soldat',
      },
```
to
```javascript
      'replaceText': {
        '\\(SE/NW\\)': '(SO/NW)',
        'center': 'mitte',
        '\\(outside\\)': '(draußen)',
        'Bug': 'Wanze',
        'Dreadnaught': 'Brummonaut',
        'Emergency Override': 'Not-Übersteuerung',
        'Knight': 'Ritter',
        'Rook': 'Drehturm',
        'Soldier': 'Soldat',
      },
```

**Comment by MaikoTan** [comment]
Thank you! I have fixed it.

**Comment by quisquous** [comment]
Thanks for this! This is great.

**Pull request merged**